src/Files.Uwp/Helpers/LayoutPreferences/LayoutPrefsDb.cs

```diff
@@ -0,0 +1,143 @@
+﻿using LiteDB;
+using System;
+using System.Linq;
+using Files.Shared.Extensions;
+
+namespace Files.Uwp.Helpers.LayoutPreferences
+{
+    public class LayoutPrefsDb : IDisposable
+    {
+        private readonly LiteDatabase db;
+
+        public LayoutPrefsDb(string connection, bool shared = false)
+        {
+            db = new LiteDatabase(new ConnectionString(connection)
+            {
+                Mode = shared ? FileMode.Shared : FileMode.Exclusive
+            }, new BsonMapper() { IncludeFields = true });
+        }
+
+        public void SetPreferences(string filePath, ulong? frn, LayoutPreferences prefs)
+        {
+            // Get a collection (or create, if doesn't exist)
+            var col = db.GetCollection<LayoutDbPrefs>("layoutprefs");
+
+            var tmp = _FindPreferences(filePath, frn);
+            if (tmp == null)
+            {
+                if (prefs != null)
+                {
+                    // Insert new tagged file (Id will be auto-incremented)
+                    var newPref = new LayoutDbPrefs
+                    {
+                        FilePath = filePath,
+                        Frn = frn,
+                        Prefs = prefs
+                    };
+                    col.Insert(newPref);
+                }
+            }
+            else
+            {
+                if (prefs != null)
+                {
+                    // Update file tag
+                    tmp.Prefs = prefs;
+                    col.Update(tmp);
+                }
+                else
+                {
+                    // Remove file tag
+                    col.Delete(tmp.Id);
+                }
+            }
+        }
+
+        public LayoutPreferences GetPreferences(string filePath = null, ulong? frn = null)
+        {
+            return _FindPreferences(filePath, frn)?.Prefs;
+        }
+
+        private LayoutDbPrefs _FindPreferences(string filePath = null, ulong? frn = null)
```

Comment:
If we assign filePath to null, maybe we should also mark it as nullable?

Reply:
Done, enabled nullable for this file.